Source/Shapes/Line.cpp

```diff
@@ -1,18 +1,17 @@
 #include "Ashkal/Shapes/Line.hpp"
-#include <climits>
 
 using namespace Ashkal;
 
-Line::Line()
-  : Line({0, 0}) {}
-
-Line::Line(const Point& point)
-  : m_point(point) {}
-
-Point Line::get_point() const {
-  return m_point;
-}
-
 void Line::accept(ShapeVisitor& visitor) const {
   visitor.visit(*this);
 }
+
+Line Ashkal::make_line(const Point& p1, const Point& p2) {
+  auto l = Line();
+  auto length = std::sqrt(std::pow(std::abs(p1.x - p2.x), 2) +
+    std::pow(std::abs(p1.y - p2.y), 2));
+  auto slope = (p1.y - p2.y) / (p1.x - p2.x);
```

Comment:
This won't work for a vertical line. Write a unit test to exercise a vertical line case and then fix the implementation.